tests/BoltOn.Tests/Data/CosmosDb/CosmosDbRepositoryTests.cs

```diff
@@ -1,52 +1,145 @@
+        {
+            // arrange
+            var id = Guid.Parse("eda6ac19-0b7c-4698-a1f7-88279339d9ff");
+
+            // act
+            var result = await _sut.GetByIdAsync(id, new RequestOptions { PartitionKey = new PartitionKey(1) });
+
+            // assert
+            Assert.NotNull(result);
+            Assert.Equal("John".ToLower(), result.FirstName.ToLower());
+        }
+
+        [Fact, Trait("Category", "Integration")]
+        public async Task GetAllAsync_WhenRecordsExist_ReturnsAllTheRecords()
+        {
+            // arrange
+
+            // act
+            var result = await _sut.GetAllAsync();
+
+            // assert
+            Assert.Single(result);
+        }
+
+        [Fact, Trait("Category", "Integration")]
+        public async Task FindByAsync_WhenRecordDoesNotExist_ReturnsNull()
+        {
+            // arrange
+
+            // act
+            var result = (await _sut.FindByAsync(f => f.StudentTypeId == 1 && f.FirstName == "johnny")).FirstOrDefault();
+
+            // assert
+            Assert.Null(result);
+        }
+
+        [Fact, Trait("Category", "Integration")]
+        public async Task FindByAsync_WhenRecordExist_ReturnsRecord()
+        {
+            // arrange
+
+            // act
+            var result = (await _sut.FindByAsync(f => f.StudentTypeId == 1 && f.FirstName == "john")).FirstOrDefault();
+
+            // assert
+            Assert.NotNull(result);
+        }
+
+        [Fact, Trait("Category", "Integration")]
+        public async Task UpdateAsync_UpdateAnExistingEntity_UpdatesTheEntity()
+        {
+            // arrange
+            var id = Guid.Parse("eda6ac19-0b7c-4698-a1f7-88279339d9ff");
+            var student = new StudentFlattened { Id = id, LastName = "smith jr", FirstName = "john", StudentTypeId = 1 };
+
+            // act
+            await _sut.UpdateAsync(student, new RequestOptions { PartitionKey = new PartitionKey(1) });
+
+            // assert
+            var result = (await _sut.FindByAsync(f => f.StudentTypeId == 1 && f.FirstName == "john")).FirstOrDefault();
+            Assert.NotNull(result);
+            Assert.Equal("smith jr".ToLower(), result.LastName.ToLower());
+        }
+
+        [Fact, Trait("Category", "Integration")]
+        public async Task AddAsync_AddANewEntity_ReturnsAddedEntity()
+        {
+            // arrange
+            var id = Guid.NewGuid();
+            var studentFlattened = new StudentFlattened { Id = id, StudentTypeId = 2, FirstName = "meghan", LastName = "doe" };
+
+            // act
+            var addedEntity = await _sut.AddAsync(studentFlattened);
+
+            // assert
+            var result = await _sut.GetByIdAsync(id, new RequestOptions { PartitionKey = new PartitionKey(2) });
+            Assert.NotNull(result);
+            Assert.Equal("meghan".ToLower(), result.FirstName.ToLower());
+            await _sut.DeleteAsync(studentFlattened, new RequestOptions { PartitionKey = new PartitionKey(2) });
```

Comment:
why deleteasync? why "meghan".ToLower()

Reply:
@gokulm : as i am always assuming collection to have one record which was added by me i am deleting it again.

Reply:
why do u assume the collection to have only one record?